### ISSUE TITLE
Authenticate users and handle sessions

### ISSUE THREAD

**Issue by gleidsonlm** [gleidsonlm/angelon-api#3]
Users should be able to create their accounts or complete the first step requesting moderator confirmation. User authenticated has its object attached in the request. Customer decorator @User() intercepts the request, confirms authentication, and adds the object and JWT.
Authentication on NestJS by [Passport](https://github.com/jaredhanson/passport). https://docs.nestjs.com/security/authentication

Example: 

```
auth.decorator.ts JS

import { applyDecorators } from '@nestjs/common';

export function Auth(...roles: Role[]) {
  return applyDecorators(
    SetMetadata('roles', roles),
    UseGuards(AuthGuard, RolesGuard),
    ApiBearerAuth(),
    ApiUnauthorizedResponse({ description: 'Unauthorized' }),
  );
}
```
from: https://docs.nestjs.com/custom-decorators

**Comment by gleidsonlm** [comment]
Resolved with Pull Request from "dev" branch #4

**Comment by gleidsonlm** [comment]
Resolved with Pull Request from "dev" branch #4
